WordPress/src/main/java/org/wordpress/android/ui/people/PeopleListFragment.java

```diff
@@ -72,16 +93,120 @@ public void refreshPeopleList() {
         }
     }
 
+    /*
+    * show/hide progress bar which appears at the bottom of the activity when loading more people
+    */
+    public void showLoadingProgress(boolean showProgress) {
+        if (isAdded() && mProgress != null) {
+            if (showProgress) {
+                mProgress.bringToFront();
+                mProgress.setVisibility(View.VISIBLE);
+            } else {
+                mProgress.setVisibility(View.GONE);
+            }
+        }
+    }
+
     @Override
     public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-        if (mListener != null) {
+        if (mOnPersonSelectedListener != null) {
             Person person = (Person) parent.getItemAtPosition(position);
-            mListener.onPersonSelected(person);
+            mOnPersonSelectedListener.onPersonSelected(person);
         }
     }
 
     // Container Activity must implement this interface
     public interface OnPersonSelectedListener {
         void onPersonSelected(Person person);
     }
+
+    public interface OnFetchMorePeopleListener {
+        void onFetchMorePeople();
+    }
+
+    public class PeopleAdapter extends BaseAdapter {
+        private final LayoutInflater mInflater;
+        private List<Person> mPeopleList;
+        private int mAvatarSz;
+
+        public PeopleAdapter(Context context, List<Person> peopleList) {
+            mAvatarSz = context.getResources().getDimensionPixelSize(R.dimen.people_avatar_sz);
+            mInflater = LayoutInflater.from(context);
+            mPeopleList = peopleList;
+        }
+
+        public void setPeopleList(List<Person> peopleList) {
+            mPeopleList = peopleList;
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public int getCount() {
+            if (mPeopleList == null) {
+                return 0;
+            }
+            return mPeopleList.size();
+        }
+
+        @Override
+        public Person getItem(int position) {
+            if (mPeopleList == null) {
+                return null;
+            }
+            return mPeopleList.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            Person person = getItem(position);
+            if (person == null) {
+                return -1;
+            }
+            return person.getPersonID();
```

Comment:
Since each person's ID is different, the list can be optimized by returning `true` for `hasStableIds()`, ex:
```
@Override
public boolean hasStableIds() {
    return true;
}
```
Explanation [here](http://stackoverflow.com/a/29446944/1673548).